persistence/src/main/java/com/epam/esm/dao/impl/TagDaoImpl.java

```diff
@@ -2,77 +2,64 @@
 
 import com.epam.esm.dao.TagDao;
 import com.epam.esm.entity.Tag;
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.jdbc.core.JdbcTemplate;
 import org.springframework.stereotype.Repository;
 
-import java.util.List;
+import javax.persistence.EntityManager;
 import java.util.Optional;
 
 @Repository
-public class TagDaoImpl extends AbstractDao<Tag> implements TagDao {
+public class TagDaoImpl extends AbstractDao<Long, Tag> implements TagDao {
 
-    private static final String TABLE_NAME = "tag";
-    private static final String ID_COLUMN_NAME = "id";
+    private static final String TOP_TAG_SQL = """
```

Comment:
Please leave a comment explaining what this query does and why it is written like this.